src/lib/components/ReactSearchKit/ReactSearchKit.js

```diff
@@ -35,15 +35,21 @@ export class ReactSearchKit extends Component {
     this.store = createStoreWithConfig(appConfig);
     this.appName = props.appName;
     this.eventListenerEnabled = props.eventListenerEnabled;
+    this.componentIndex = 0;
+  }
+  nextComponentIndex(rc) {
+    /* This is an internal function that will be used to create id for the checkboxes and labels, ensuring that
+     they are unique */
+    return rc.componentIndex++;
   }
-
   render() {
     const { appName, children, eventListenerEnabled, overridableId, searchOnInit } =
       this.props;
 
     const context = {
       appName,
       buildUID: (element, overrideId) => buildUID(element, overrideId, appName),
+      nextComponentIndex: () => this.nextComponentIndex(this),
```

Comment:
Please, check this! I started with something like :
```
  nextComponentIndex: this.nextComponentIndex,
```
and that failed because then `this` was empty inside the `nextComponentIndex` function. If there is a better way of making sure that `this` is available, please, let me know

Reply:
There are two things happing here:
- the first is the always confusing usage of `this` in js
- the second is that the way you are testing (by changing the code in the compiled version of this JS) **might** differ from what the JS/babel compiler will do. The usage of `this` might be compiled differently. It might be the same, but it might be not.

To be sure, you can run the demo apps inside React-SearchKit, following the README, so you have a local running app to test.

Given said so, I think that your previous version should work:
```javascript
getNextUniqueId() {
    return `${this.appName}-${this.nextUniqueId++}`;
}
```

Reply:
Fixed, thanks